core/src/apps/management/apply_settings.py

```diff
@@ -12,6 +12,21 @@
     from trezor.messages.ApplySettings import ApplySettings, EnumTypeSafetyCheckLevel
 
 
+def validate_homescreen(homescreen: bytes) -> None:
+    if homescreen == b"":
+        return
+
+    if len(homescreen) > storage.device.HOMESCREEN_MAXSIZE:
+        raise wire.DataError("Homescreen is too complex")
+
+    try:
+        w, h, grayscale = ui.display.toif_info(homescreen)
+    except ValueError:
+        raise wire.DataError("Invalid homescreen")
+    if w != 144 or h != 144 or grayscale:
+        raise wire.DataError("Invalid homescreen")
```

Comment:
dttto. Maybe `Homescreen image needs to be 144 x 144.` and `Homescreen image can't be grayscale` (split it into two).

Reply:
Fixed in 39d0f79a.